rpc_jobs/phobos.yml

```diff
@@ -0,0 +1,194 @@
+              )
+              // MAAS_TOKEN is added to env in "Check MaaS Credentials"
+          ]){
+            // do checkout before warning users to ensure branch is valid.
+            stage("Checkout"){
+              dir("rpc-eng-ops"){
+                git branch: env.RPC_ENG_OPS_BRANCH, url: env.RPC_ENG_OPS_REPO, credentialsId: 'rpc-jenkins-svc-github-ssh-key'
+              }
+              // Shell used instead of git function as I want to make all
+              // modifications to the exisiting clone expicit
+              sh """#!/bin/bash -xeu
+                cd /opt/rpc-openstack
+                # log current HEAD
+                git show
+                git status
+
+                pushd openstack-ansible
+                  sudo git stash
+                  sudo git reset --hard
+                popd
+                sudo git fetch --all
+                sudo git stash
+                sudo git reset --hard
+                sudo git checkout "${env.RPC_BRANCH}"
+                sudo git reset --hard origin/${env.RPC_BRANCH}
+                sudo git submodule update --init
+
+                cd /opt/rpc-maas
+                sudo git fetch --all
+                sudo git stash
+                sudo git reset --hard
+                sudo git checkout ${env.RPC_MAAS_BRANCH}
+                sudo git reset --hard origin/${env.RPC_MAAS_BRANCH}
+              """
+            }
+            stage("Check MaaS Credentials"){
+              // MaaS credentials check must be performed on an internal
+              // slave as it may use the internal identity api which
+              // isn't available from phobos.
+              common.internal_slave(){
+                dir("rpc-eng-ops"){
+                  git branch: env.RPC_ENG_OPS_BRANCH, url: env.RPC_ENG_OPS_REPO, credentialsId: 'rpc-jenkins-svc-github-ssh-key'
+                }
+                sh """#!/bin/bash
+                  scl enable python27 ${WORKSPACE}/rpc-eng-ops/phobos-re-automation/maas_creds_check.sh
+                """
+                // Ensure that the MAAS_TOKEN env var is correct
+                // whether the token was obtained from the service account
+                // or supplied via job param.
+                env.MAAS_TOKEN = readFile file: "MAAS_TOKEN"
+              }
+            }
+            stage("Pre Deployment Verify"){
+              verify()
+            }
+            stage("Pre Deploy Warning"){
+                if (env.PRE_DEPLOY_PAUSE == "true"){
+                  slack_notify("Phobos Deploy starting in 15 minutes.", "warning")
+                  sleep(time: 15, unit: "MINUTES")
+                }
+                slack_notify("Phobos Deploy starting now.", "warning")
+            }
+            stage("Deploy"){
+              if (env.DEPLOY == "true"){
+                sh """#!/bin/bash -xeu
+                  sudo -E rpc-eng-ops/phobos-re-automation/deploy.sh
```

Comment:
Has the env been (re)deployed with this tool already?

Reply:
rcbops/rpc-eng-ops#123 is listed as a dep in the PR description. But no, the deploy script has not been run, I wanted it to be reviewed before its executed against the live cluster.